src/main/java/com/citihub/configr/namespace/NamespaceService.java

```diff
@@ -78,6 +92,45 @@ public Namespace storeNamespace(Map<String, Object> json, String path, boolean s
     }
   }
 
+  private void validateNamespace(Namespace namespace)
+      throws SchemaValidationException, JsonProcessingException {
+
+    // TODO: Should we use metadataService.getMetadataForNamespace ?
+    // or will namespace.getMetadata() perform lazy loading?
+    Metadata metadata = namespace.getMetadata();
+    if (metadata == null) {
+      // TODO: Do we ignore validation if there is no metadata? This could
+      // happen during initial insertion.
+      return;
+    }
+
+    ValidationLevel validationLevel = metadata.getValidationLevel();
+    String schema = metadata.getSchema();
+
+    if (validationLevel == ValidationLevel.NONE) {
+      return;
+    }
+
+    SchemaValidationResult result = this.schemaValidationService
+        .validateJSON(objectMapper.writeValueAsString(namespace.getValue()), schema);
+
+
+    switch (validationLevel) {
+      case STRICT:
+        if (!result.isSuccess()) {
+          throw new SchemaValidationException(result.getMessage());
+        }
+        break;
+      case LOOSE:
+        // TODO: What should we do here?
```

Comment:
let's set a custom response header, `X-Schema-Validation` and just do something basic, e.g. "{ \"valid\": false }" - i've staged adding an endpoint that will let you get the full validation report back